src/Core/Managed/Shared/Extensibility/Implementation/DebugOutput.cs

```diff
@@ -8,11 +8,41 @@ namespace Microsoft.ApplicationInsights.Extensibility.Implementation
 {
     using System;
     using System.Diagnostics;
+    using Channel;
     using Microsoft.ApplicationInsights.Extensibility;
-
-    internal class DebugOutput : IDebugOutput
+   
+    /// <summary>
+    /// Writes telemetry items to debug output.
+    /// </summary>
+    public class DebugOutput : IDebugOutput
     {
-        public void WriteLine(string message)
+        /// <summary>
+        /// Write the specified <see cref="ITelemetry"/> item to debug output.
+        /// </summary>
+        /// <param name="telemetry">Item to write.</param>
+        /// <param name="filteredBy">If specified, indicates the telemetry item was filtered out and not sent to the API.</param>
+        public void WriteTelemetry(ITelemetry telemetry, string filteredBy = null)
```

Comment:
I'm open to suggestions on other places to put this method. It does need to be publicly exposed though, so the telemetry processors can access it.